### PR TITLE
Fix edge case bug in the slashing logic

### PR DESCRIPTION
**Pull request by vtleonardo**
## Scope

This PR is:
- Fixing the error where the validators were not being properly evicted after a minor slash in case the disputeRewards was equal to the stakeAmount (edge case)
- Fix unit tests that were failing due to race conditions 
- Add more unit tests and improve the existing ones around the ValidatorPool contract
- Replace old evm call to progress the hardhat chain during the unit tests with the new [hardhat_mine](hardhat_mine) to speed up the unit tests
- Add requirement of node version 16 in the package.json file

### CODE REVIEW

File: bridge/contracts/ValidatorPool.sol [hunk 1]
```diff
@@ -268,7 +281,11 @@ contract ValidatorPool is
             stakeTokenID = _mintStakeNFTPosition(minerShares);
             _moveToExitingQueue(dishonestValidator_, stakeTokenID);
         } else {
-            _removeExitingQueueData(dishonestValidator_);
+            if (isValidator(dishonestValidator_)) {
```
Comment (vtleonardo) [inline bridge/contracts/ValidatorPool.sol:284]: The edge case was fixed here!

File: bridge/contracts/ValidatorPool.sol [hunk 2]
```diff
@@ -255,6 +264,10 @@ contract ValidatorPool is
     }
 
     function minorSlash(address dishonestValidator_, address disputer_) public onlyETHDKG {
+        require(
```
Comment (vtleonardo) [inline bridge/contracts/ValidatorPool.sol:267]: Adding more requirements to the slash functions (being a little bit more paranoic)

File: bridge/test/validatorPool/business-logic/claim-position.test.ts [hunk 3]
```diff
@@ -134,49 +120,6 @@ describe("ValidatorPool: Claiming logic", async () => {
     }
   });
 
-  it("After the claim period, the user should be able to claim its stakenft position", async function () {
```
Comment (vtleonardo) [inline bridge/test/validatorPool/business-logic/claim-position.test.ts:137]: This unit test was a duplicate, it was doing the same as some tests above.
